src/rng.rs

```diff
@@ -91,20 +82,21 @@ impl FiatShamirRngSeed for FiatShamirChaChaRngSeed {
     fn add_bytes<'a, T: 'a + ToBytes>(&mut self, elem: &'a T) -> Result<&mut Self, Self::Error> {
 
         // Check we have not reached the maximum allowed seed size
-        if self.num_elements as usize == Self::MAX_NUM_ELEMENTS {
-            return Err(Error::BadFiatShamirInitialization(format!("Maximum seed length {} exceeded", Self::MAX_NUM_ELEMENTS)));
+        if self.num_elements == u64::MAX {
+            return Err(Error::BadFiatShamirInitialization(format!("Maximum seed length {} exceeded", u64::MAX)));
         }
 
         // Get elem bytes and check that they are not over the maximum allowed elem len
-        let mut elem_bytes = to_bytes!(elem).map_err(|_| Error::BadFiatShamirInitialization("Unable to convert elem to bytes".to_owned()))?;
-        let elem_bytes_len = elem_bytes.len();
-        if elem_bytes_len > Self::MAX_ELEMENT_LEN {
-            return Err(Error::BadFiatShamirInitialization(format!("Max elem length exceeded. Max: {}, Elem len: {}", Self::MAX_ELEMENT_LEN, elem_bytes_len)));
-        }
+        let mut elem_bytes = to_bytes!(elem).map_err(
+            |_| Error::BadFiatShamirInitialization("Unable to convert elem to bytes".to_owned())
+        )?;
+        let elem_bytes_len: u64 = elem_bytes.len().try_into().map_err(
+            |_| Error::BadFiatShamirInitialization(format!("Max elem length exceeded. Max: {}", u64::MAX))
+        )?;
 
         // Update internal state
         self.num_elements += 1;
-        self.elements_len.push(elem_bytes_len as u64);
+        self.elements_len.push(elem_bytes_len);
```

Comment:
We should maybe put a comment on our encoding above ``fn add_bytes``.
We use 8 Bytes for the length of the "payload", and then the "payload" itself.